### PR TITLE
Make 'ViForwardChar' able to accept suggestions

### PR DESCRIPTION
**Pull request by daxian-dbw**
- Make `ViForwardChar` able to accept suggestions
- Fix the cursor position in `Undo` in VI mode, so the cursor is correctly placed.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/PowerShell/PSReadLine/pull/1528)

### CODE REVIEW

File: PSReadLine/UndoRedo.cs [hunk 1]
```diff
@@ -75,9 +75,10 @@ public static void Undo(ConsoleKeyInfo? key = null, object arg = null)
                 }
                 _singleton._edits[_singleton._undoEditIndex - 1].Undo();
                 _singleton._undoEditIndex--;
+
                 if (_singleton._options.EditMode == EditMode.Vi && _singleton._current >= _singleton._buffer.Length)
                 {
-                    _singleton._current = Math.Max(0, _singleton._buffer.Length - 1);
+                    _singleton._current = Math.Max(0, _singleton._buffer.Length + ViEndOfLineFactor);
```
Comment (lzybkr) [inline PSReadLine/UndoRedo.cs:81]: Is this change related? Is it tested?

Comment (daxian-dbw) [reply PSReadLine/UndoRedo.cs:81]: It's related. It's tested by [line 318](https://github.com/PowerShell/PSReadLine/pull/1528/files#diff-588be69a09431df5cd53b7808a683dc5R318) and [line 324](https://github.com/PowerShell/PSReadLine/pull/1528/files#diff-588be69a09431df5cd53b7808a683dc5R324) for insert mode and command mode respectively.

**Review by lzybkr — APPROVED** [review]
